estimagic/logging/create_database.py

```diff
@@ -0,0 +1,253 @@
+    gradient_status=0,
+    replace=False,
+):
+    """Return database metadata object with all relevant tables for the optimization.
+
+    This should always be used to create entirely new databases or to create the
+    tables needed during optimization in an existing database.
+
+    A new database is created if path does not exist yet or replace=True. Otherwise the
+    existing database is loaded and all tables needed to log the optimization are
+    overwritten.
+
+    The resulting database has the following tables:
+
+    - params_history: the complete history of parameters from the optimization. The
+      index column is "iteration", the remaining columns are parameter names taken
+      from params["name"].
+    - gradient_history: the complete history of gradient evaluations from the
+      optimization. Same columns as params_history.
+    - criterion_history: the complete history of criterion values from the optimization.
+      The index column is "iteration", the second column is "value".
+    - time_stamps: timestamps from the end of each criterion evaluation. Same columns as
+      criterion_history.
+    - convergence_history: the complete history of convergence criteria from the
+      optimization. The index column is "iteration", the other columns are "ftol",
+      "gtol" and "xtol".
+    - start_params: copy of user provided ``params``. This is not just the fist entry of
+      params_history because it contains all columns and has a different index.
+    - optimization_status: table with one row and one column called "value" which takes
+      the values "scheduled", "running", "success" or "failure". Initialized to
+      ``optimization_status``.
+    - gradient_status: table with one row and one column called "value" which takes
+      can be any float between 0 and 1 and indicates the progress of the gradient
+      calculation. Initialized to ``gradient_status``
+    - db_options: table with one row and one column called "value". It contains
+      a dictionary with the dashboard options. Internally this is a PickleType, so the
+      dictionary must be pickle serializable. Initialized to db_options.
+
+
+    Args:
+        path (str or pathlib.Path): location of the database file. If the file does
+            not exist, it will be created.
+        params (pd.DataFrame): see :ref:`params`
+        db_options (dict): Dashboard options.
+        optimization_status (str): One of "scheduled", "running", "success", "failure".
+        gradient_status (float): Progress of gradient calculation between 0 and 1.
+        replace (bool): If true and the database exists, it will be overwritten.
+            Otherwise, data will be appended.
+
+    Returns:
+        database (sqlalchemy.sql.schema.MetaData). The engine that connects
+        to the database can be accessed via ``database.bind``.
+
+    """
+    db_options = {} if db_options is None else db_options
+    gradient_status = float(gradient_status)
```

Comment:
Why float?

Reply:
This is a percentage used to make a progress bar.